presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArrayArithmeticFunctions.java

```diff
@@ -52,4 +52,30 @@ public static String arrayAverage()
                 "(s, x) -> IF(x IS NOT NULL, (s[1] + x, s[2] + 1), s), " +
                 "s -> if(s[2] = 0, cast(null as double), s[1] / cast(s[2] as double)))";
     }
+
+    @SqlInvokedScalarFunction(value = "array_frequency", deterministic = true, calledOnNullInput = false)
+    @Description("Returns the frequency of all array elements as a map.")
+    @SqlParameter(name = "input", type = "array<Bigint>")
+    @SqlType("MAP<Bigint, int>")
+    public static String arrayFrequencyBigint()
+    {
+        return "RETURN reduce(" +
+                "input," +
+                "MAP()," +
+                "(m, x) -> IF (x IS NOT NULL, MAP_CONCAT(m,MAP_FROM_ENTRIES(ARRAY[ROW(x, COALESCE(ELEMENT_AT(m,x) + 1, 1))])), m)," +
+                "m -> m)";
+    }
+
+    @SqlInvokedScalarFunction(value = "array_frequency", deterministic = true, calledOnNullInput = false)
+    @Description("Returns the frequency of all array elements as a map.")
+    @SqlParameter(name = "input", type = "array<Varchar>")
+    @SqlType("MAP<Varchar, int>")
```

Comment:
lowercase `map(varchar, int)`

Generally we use `()` rather than `<>` in type signature.